File: app/src/main/java/life/plank/juna/zone/view/adapter/fixture/FixtureMatchdayAdapter.java
package life.plank.juna.zone.view.adapter.fixture;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.lang.ref.WeakReference;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import life.plank.juna.zone.R;
import life.plank.juna.zone.data.model.FixtureByMatchDay;
import life.plank.juna.zone.interfaces.LeagueContainer;
import life.plank.juna.zone.util.BaseRecyclerView;
import life.plank.juna.zone.view.fragment.football.LeagueInfoFragment;

import static life.plank.juna.zone.util.AppConstants.TODAY_MATCHES;
import static life.plank.juna.zone.util.DataUtil.findString;
import static life.plank.juna.zone.util.DataUtil.isNullOrEmpty;
import static life.plank.juna.zone.util.UIDisplayUtil.findColor;

/**
 * Created by plank-prachi on 4/10/2018.
 */
public class FixtureMatchdayAdapter extends BaseRecyclerView.Adapter<FixtureMatchdayAdapter.FixtureMatchDayViewHolder> {

    private LeagueContainer leagueContainer;
    private List<FixtureByMatchDay> fixtureByMatchDayList;

    public FixtureMatchdayAdapter(LeagueContainer leagueContainer) {
        this.leagueContainer = leagueContainer;
        fixtureByMatchDayList = LeagueInfoFragment.Companion.getFixtureByMatchDayList();
    }

    @NonNull
    @Override
    public FixtureMatchDayViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new FixtureMatchDayViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(R.layout.item_fixture_matchday, parent, false),
                this
        );
    }

    @Override
    public int getItemCount() {
        return isNullOrEmpty(fixtureByMatchDayList) ? 0 : fixtureByMatchDayList.size();
    }

    public void updateFixtures() {
        fixtureByMatchDayList = LeagueInfoFragment.Companion.getFixtureByMatchDayList();
        notifyDataSetChanged();
    }

    public static class FixtureMatchDayViewHolder extends BaseRecyclerView.ViewHolder {

        private final WeakReference<FixtureMatchdayAdapter> ref;
        @BindView(R.id.matchday_header)
        TextView matchdayHeader;
        @BindView(R.id.fixtures_matchday_list)
        RecyclerView recyclerView;

        FixtureMatchDayViewHolder(View itemView, FixtureMatchdayAdapter adapter) {
            super(itemView);
            this.ref = new WeakReference<>(adapter);
            ButterKnife.bind(this, itemView);
        }

        @Override
        public void bind() {
            FixtureByMatchDay fixtureByMatchDay = ref.get().fixtureByMatchDayList.get(getAdapterPosition());

            if (!isNullOrEmpty(fixtureByMatchDay.getFixtureByDateList())) {
                matchdayHeader.setBackgroundResource(
                        fixtureByMatchDay.getDaySection().equals(TODAY_MATCHES) ?
                                R.color.light_header :
                                R.color.lighter_header
                );
                matchdayHeader.setTextColor(findColor(
                        fixtureByMatchDay.getDaySection().equals(TODAY_MATCHES) ?
                                R.color.fab_button_pink :
                                R.color.grey
                ));
                String matchdayHeaderText =
                        findString(ref.get().leagueContainer.getTheLeague().isCup() ? R.string.round_ : R.string.matchday_) +
                                (ref.get().leagueContainer.getTheLeague().isCup() ? fixtureByMatchDay.getMatchDay() : fixtureByMatchDay.getMatchDay());
                matchdayHeader.setText(matchdayHeaderText);
                recyclerView.setAdapter(new FixtureDateAdapter(fixtureByMatchDay.getFixtureByDateList(), ref.get().leagueContainer));
            }
        }
    }
}

File: app/src/main/java/life/plank/juna/zone/view/adapter/fixture/FixtureMatchdayAdapter.kt
package life.plank.juna.zone.view.adapter.fixture

import android.view.*
import androidx.recyclerview.widget.RecyclerView
import com.prembros.facilis.util.isNullOrEmpty
import kotlinx.android.synthetic.main.item_fixture_matchday.view.*
import life.plank.juna.zone.R
import life.plank.juna.zone.data.model.FixtureByDate
import life.plank.juna.zone.interfaces.LeagueContainer
import life.plank.juna.zone.util.common.AppConstants.TODAY_MATCHES
import life.plank.juna.zone.util.common.JunaDataUtil.findString
import life.plank.juna.zone.util.view.UIDisplayUtil.findColor
import life.plank.juna.zone.view.fragment.football.LeagueInfoFragment
import life.plank.juna.zone.view.fragment.football.LeagueInfoFragment.Companion.fixtureByMatchDayList

class FixtureMatchdayAdapter(private val leagueContainer: LeagueContainer) : RecyclerView.Adapter<FixtureMatchdayAdapter.FixtureMatchDayViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FixtureMatchDayViewHolder {
        return FixtureMatchDayViewHolder(
                LayoutInflater.from(parent.context).inflate(R.layout.item_fixture_matchday, parent, false)
        )
    }

    override fun onBindViewHolder(holder: FixtureMatchDayViewHolder, position: Int) {
        val (matchDay, daySection, fixtureByDateList) = fixtureByMatchDayList[position]

        holder.itemView.run {
            if (!isNullOrEmpty<FixtureByDate>(fixtureByDateList)) {
                matchday_header.setBackgroundResource(
                        if (daySection == TODAY_MATCHES)
                            R.color.light_header
                        else
                            R.color.lighter_header
                )
                matchday_header.setTextColor(findColor(
                        if (daySection == TODAY_MATCHES)
                            R.color.fab_button_pink
                        else
                            R.color.grey
                ))
                val matchdayHeaderText = findString(if (leagueContainer.getTheLeague().isCup) R.string.round_ else R.string.matchday_) + if (leagueContainer.getTheLeague().isCup) matchDay else matchDay
                matchday_header.text = matchdayHeaderText
                fixtures_matchday_list.adapter = FixtureDateAdapter(fixtureByDateList, leagueContainer)
            }
        }
    }

    override fun getItemCount(): Int = if (isNullOrEmpty(fixtureByMatchDayList)) 0 else fixtureByMatchDayList.size

    fun updateFixtures() {
        fixtureByMatchDayList = LeagueInfoFragment.fixtureByMatchDayList
        notifyDataSetChanged()
    }

    class FixtureMatchDayViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}